providers/implementations/encode_decode/ml_kem_codecs.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef PROV_ML_KEM_CODECS_H
+# define PROV_ML_KEM_CODECS_H
+# pragma once
+
+# include <openssl/e_os2.h>
+# include "crypto/ml_kem.h"
+# include "prov/provider_ctx.h"
+
+ /*-
+  * The DER ASN.1 encoding of ML-KEM (and ML-DSA) public keys prepends 22 bytes
+  * to the encoded public key:
+  *
+  * - 4 byte outer sequence tag and length
+  * -  2 byte algorithm sequence tag and length
+  * -    2 byte algorithm OID tag and length
+  * -      9 byte algorithm OID (from NIST CSOR OID arc)
+  * -  4 byte bit string tag and length
+  * -    1 bitstring lead byte
+  */
+# define ML_KEM_SPKI_OVERHEAD   22
+typedef struct {
+    const uint8_t asn1_prefix[ML_KEM_SPKI_OVERHEAD];
+} ML_KEM_SPKI_INFO;
+
+/*-
+ * For each algorithm we support a few PKCS#8 input formats,
```

Comment:
Maybe define this first...

PrivateKey ::= SEQUENCE {
  seed OCTET STRING OPTIONAL,
  expandedKey [1] IMPLICIT OCTET STRING OPTIONAL                                                                                                                                               }
SInce the first 3 are just this definition

Reply:
As a comment above the tables, mentioning the correspondence?